File: Easy/Evaluate Expression Tree/Evaluate Expression Tree.cpp
using namespace std;

class BinaryTree
{
public:
    int value;
    BinaryTree *left = nullptr;
    BinaryTree *right = nullptr;

    BinaryTree(int value) { this->value = value; }
};

int evaluateExpressionTree(BinaryTree *tree)
{
    if (tree->left == nullptr && tree->right == nullptr)
        return tree->value;
    switch (tree->value)
    {
    case -1:
        return evaluateExpressionTree(tree->left) + evaluateExpressionTree(tree->right);
    case -2:
        return evaluateExpressionTree(tree->left) - evaluateExpressionTree(tree->right);
    case -3:
        return evaluateExpressionTree(tree->left) / evaluateExpressionTree(tree->right);
    case -4:
        return evaluateExpressionTree(tree->left) * evaluateExpressionTree(tree->right);
    }
}

File: Easy/Evaluate Expression Tree/Evaluate Expression Tree.py
class BinaryTree:
    def __init__(self, value, left=None, right=None):
        self.value = value
        self.left = left
        self.right = right


def evaluateExpressionTree(tree):
    if not tree.left and not tree.right:
        return tree.value
    if tree.value == -1:
        return evaluateExpressionTree(tree.left) + evaluateExpressionTree(tree.right)
    elif tree.value == -2:
        return evaluateExpressionTree(tree.left) - evaluateExpressionTree(tree.right)
    elif tree.value == -3:
        return int(evaluateExpressionTree(tree.left) / evaluateExpressionTree(tree.right))
    elif tree.value == -4:
        return evaluateExpressionTree(tree.left) * evaluateExpressionTree(tree.right)
